File: src/main/java/com/example/payheurebackend/service/PointageAnomalieService.java
package com.example.payheurebackend.service;

import com.example.payheurebackend.dto.PointageAnomalieRequest;
import com.example.payheurebackend.dto.PointageAnomalieResponse;

import java.util.List;

/** Recherche, tous salariés confondus, des pointages incomplets ou oubliés sur une période. */
public interface PointageAnomalieService {

    /**
     * Reconstitue les sessions de travail de chaque salarié actif sur la période demandée et ne
     * renvoie que ceux ayant au moins un badgeage sans sortie correspondante.
     *
     * @throws com.example.payheurebackend.exception.InvalidPeriodException si la période est incohérente
     */
    List<PointageAnomalieResponse> lister(PointageAnomalieRequest request);
}

File: PayHeureUI/src/app/services/pointage-anomalie.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { PointageAnomalieRequest, PointageAnomalieResponse } from '../models/PointageAnomalie';
import { API_BASE_URL } from '../config/api.config';

@Injectable({
  providedIn: 'root'
})
export class PointageAnomalieService {

  private readonly apiUrl = `${API_BASE_URL}/pointages`;

  constructor(private http: HttpClient) {}

  /** Salariés ayant, sur la période demandée, au moins un badgeage sans sortie correspondante. */
  lister(request: PointageAnomalieRequest): Observable<PointageAnomalieResponse[]> {
    return this.http.post<PointageAnomalieResponse[]>(`${this.apiUrl}/anomalies`, request);
  }
}
